ocs_ci/deployment/ocp.py

```diff
@@ -0,0 +1,177 @@
+            msg = "The given cluster path is not empty: {}. ".format(
+                self.cluster_path)
+            msg += (
+                f"Provide an empty --cluster-path and --deploy to deploy "
+                f"a new cluster"
+            )
+            pytest.fail(msg)
+        else:
+            logger.info(
+                f"A testing cluster will be deployed and cluster information "
+                f"stored at: %s",
+                self.cluster_path
+            )
+        self.create_config()
+
+    def create_config(self):
+        """
+        Create the OCP deploy config, if something needs to be changed fro
+        specific platform you can overload this method in child class.
+        """
+        # Generate install-config from template
+        logger.info("Generating install-config")
+        _templating = templating.Templating()
+        ocp_install_template = (
+            f"install-config-{self.deployment_platform}-"
+            f"{self.deployment_type}.yaml.j2"
+        )
+        ocp_install_template_path = os.path.join(
+            "ocp-deployment", ocp_install_template
+        )
+        install_config_str = _templating.render_template(
+            ocp_install_template_path, config.ENV_DATA
+        )
+        # Log the install config *before* adding the pull secret,
+        # so we don't leak sensitive data.
+        logger.info(f"Install config: \n{install_config_str}")
+        # Parse the rendered YAML so that we can manipulate the object directly
+        install_config_obj = yaml.safe_load(install_config_str)
+        install_config_obj['pullSecret'] = self.get_pull_secret()
+        install_config_str = yaml.safe_dump(install_config_obj)
+        install_config = os.path.join(self.cluster_path, "install-config.yaml")
+        with open(install_config, "w") as f:
+            f.write(install_config_str)
+
+    def deploy(self, log_cli_level='DEBUG'):
+        """
+        Implement ocp deploy in specific child class
+        """
+        raise NotImplementedError("deploy_ocp functionality not implemented")
+
+    def test_cluster(self):
+        """
+        Test if OCP cluster installed successfuly
+        """
+        # Test cluster access
+        if not OCP.set_kubeconfig(
+            os.path.join(
+                self.cluster_path, config.RUN.get('kubeconfig_location'),
+            )
+        ):
+            pytest.fail("Cluster is not available!")
+
+    def destroy(self, log_level="DEBUG"):
+        """
+        Destroy OCP cluster specific
+
+        Args:
+            log_level (str): log level openshift-installer (default: DEBUG)
+        """
+        logger.info("Destroying the cluster")
+        destroy_cmd = (
+            f"{self.installer} destroy cluster "
+            f"--dir {self.cluster_path} "
+            f"--log-level {log_level}"
+        )
+
+        try:
+
+            # Retrieve cluster name and AWS region from metadata
```

Comment:
This shouldn't be specific for AWS, right? (probably just the comment mentions it incorrectly)